keentools/utils/points.py

```diff
@@ -127,26 +150,33 @@ def get_add_selection_mode(self) -> bool:
         return self._add_selection_mode
 
     def on_start(self) -> None:
+        _log.output('on_start')
         self.set_add_selection_mode(False)
         self.set_move_pin_mode(False)
         self.reset_current_pin()
         self.clear_selected_pins()
         self.clear_disabled_pins()
 
     def clear_all(self) -> None:
-        self.on_start()
-        self.set_pins([])
+        _log.output('clear_all')
+        self.set_add_selection_mode(False)
+        self.set_move_pin_mode(False)
+        self.clear_pins()
 
     def remove_pin(self, index: int) -> None:
-        if index in self._selected_pins:
-            self._selected_pins.remove(index)
-        self._selected_pins = [x if x < index else x - 1
-                               for x in self._selected_pins]
+        _log.output(f'remove_pin: {index}')
+        pins = self.arr()
+        if index >= len(pins):
+            return
+        self._selected_pins = np.array([x if x < index else x - 1
+                                        for x in self._selected_pins
+                                        if x != index], dtype=np.int32)
         if index in self._disabled_pins:
```

Comment:
this check and removing are probably not necessary (`index` will be filtered out later)